libs/datamodel/core/src/walkers.rs

```diff
@@ -248,6 +248,20 @@ impl<'a> RelationFieldWalker<'a> {
     pub fn arity(&self) -> FieldArity {
         self.get().arity
     }
+    #[allow(clippy::needless_lifetimes)]
+    pub fn scalar_arities<'b>(&'b self) -> impl Iterator<Item = FieldArity> + 'b {
```

Comment:
So I thought clippy was just wrong before, but I think this can be written like this and make clippy happy:

```rust
pub fn scalar_arities(&self) -> impl Iterator<Item = FieldArity> + '_ {
```

Reply:
Ah, very good. I did not try that one